### PR TITLE
ci: upgrade to latest k8s version v1.25.0 for ci test

### PR DESCRIPTION
**Pull request by subhamkrai**
integration test now run on the latest k8s v1.25.0.

Signed-off-by: subhamkrai <srai@redhat.com>

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure)).
- [ ] **Skip Tests for Docs**: If this is only a documentation change, add the label `skip-ci` on the PR.
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

**Comment by subhamkrai** [comment]
not changing the version for the canary test, since multus will fail with the latest k8s which will be fixed in another pr

**Comment by subhamkrai** [comment]
all the 1.25.0 tests are failing because of PSP

**Comment by travisn** [comment]
> all the 1.25.0 tests are failing because of PSP

One of the PSPs must have been missed in #10797, but we better fix it in a separate PR so we can backport it, after it's validated in this PR.

**Comment by subhamkrai** [comment]
@mergify rebase

**Comment by travisn** [comment]
> > all the 1.25.0 tests are failing because of PSP
> 
> One of the PSPs must have been missed in #10797, but we better fix it in a separate PR so we can backport it, after it's validated in this PR.

@BlaineEXE Looks like the psp wasn't removed from common.yaml in #10797, if you could take a look

**Comment by subhamkrai** [comment]
> upgrade to k8s v1.25.0 is required in canary tests too. right? @travisn do you want to add that too or I should add a commit?

btw I have pr opened for that only https://github.com/rook/rook/pull/10536 looking we canary version can be updated there.

**Comment by travisn** [comment]
> > upgrade to k8s v1.25.0 is required in canary tests too. right? @travisn do you want to add that too or I should add a commit?
> 
> btw I have pr opened for that only #10536 looking we canary version can be updated there.

Sounds good if you update the canary tests in #10536 and we go ahead and merge this, thanks

**Review by travisn — APPROVED** [review]

File: .mergify.yml [hunk 1]
```diff
@@ -159,12 +159,12 @@ pull_request_rules:
       - "check-success=multi-cluster-mirroring"
       - "check-success=rgw-multisite-testing"
       - "check-success=TestCephSmokeSuite (v1.17.17)"
-      - "check-success=TestCephSmokeSuite (v1.24.1)"
```
[the remark this answers is not in the document]
Comment (travisn) [reply .mergify.yml]: Actually I'd like to backport this to 1.10 since we're changing the min supported version with #10809.

**Pull request merged**